app/src/main/java/com/android/nikhil/worldnow/data/source/datasource/NewsDataSource.kt

```diff
@@ -0,0 +1,13 @@
+package com.android.nikhil.worldnow.data.source.datasource
+
+import com.android.nikhil.worldnow.data.model.main_news.Result
+
+interface NewsDataSource {
+    fun getNews(callback : getNewsCallback)
+
+    interface getNewsCallback {
+        fun onNewsLoaded(data : List<Result>?)
+        fun onDatanotAvalaible()
```

Comment:
Just a minor, please fix the camel case on this method from `onDatanotAvailable()` to `onDataNotAvailable()`, it reads better.